Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -4346,431 +4346,305 @@ void CPhysicalGeometry::PrepareAdjacency(CConfig *config) {
-            b[iDim] = 0.5*(Coord_3[iDim]-Coord_1[iDim]);
-          }
-          test = a[0]*b[1]-b[0]*a[1];
+    SU2_OMP_FOR_DYN(OMP_MIN_SIZE)
+    for (auto iElem_Surface = 0ul; iElem_Surface < nElem_Bound[iMarker]; iElem_Surface++) {
 
-          if (test < 0.0) {
-            bound[iMarker][iElem_Surface]->Change_Orientation();
-            line_flip++;
-          }
+      /*--- Pick a reference point inside the domain that is not part of the surface element. ---*/
+      const auto iElem_Domain = bound[iMarker][iElem_Surface]->GetDomainElement();
+      unsigned long Point_Domain = 0;
 
+      for (auto iNode_Domain = 0u; iNode_Domain < elem[iElem_Domain]->GetnNodes(); iNode_Domain++) {
+        Point_Domain = elem[iElem_Domain]->GetNode(iNode_Domain);
+        bool find = false;
+        for (auto iNode_Surface = 0u; iNode_Surface < bound[iMarker][iElem_Surface]->GetnNodes(); iNode_Surface++) {
+          auto Point_Surface = bound[iMarker][iElem_Surface]->GetNode(iNode_Surface);
+          if (Point_Surface == Point_Domain) {find = true; break;}
         }
+        if (!find) break;
+      }
 
-        /*--- 3D grid, triangle case ---*/
+      /*--- 2D grid. ---*/
 
-        if (bound[iMarker][iElem_Surface]->GetVTK_Type() == TRIANGLE) {
+      if (bound[iMarker][iElem_Surface]->GetVTK_Type() == LINE) {
 
-          Point_1_Surface = bound[iMarker][iElem_Surface]->GetNode(0); Coord_1 = nodes->GetCoord(Point_1_Surface);
-          Point_2_Surface = bound[iMarker][iElem_Surface]->GetNode(1); Coord_2 = nodes->GetCoord(Point_2_Surface);
-          Point_3_Surface = bound[iMarker][iElem_Surface]->GetNode(2); Coord_3 = nodes->GetCoord(Point_3_Surface);
-          Coord_4 = nodes->GetCoord(Point_Domain);
+        auto Point_1_Surface = bound[iMarker][iElem_Surface]->GetNode(0);
+        auto Point_2_Surface = bound[iMarker][iElem_Surface]->GetNode(1);
+        auto Coord_1 = nodes->GetCoord(Point_1_Surface);
+        auto Coord_2 = nodes->GetCoord(Point_2_Surface);
+        auto Coord_3 = nodes->GetCoord(Point_Domain);
 
-          for (iDim = 0; iDim < nDim; iDim++) {
-            a[iDim] = 0.5*(Coord_2[iDim]-Coord_1[iDim]);
-            b[iDim] = 0.5*(Coord_3[iDim]-Coord_1[iDim]);
-            c[iDim] = Coord_4[iDim]-Coord_1[iDim];
-          }
-          n[0] = a[1]*b[2]-b[1]*a[2];
-          n[1] = -(a[0]*b[2]-b[0]*a[2]);
-          n[2] = a[0]*b[1]-b[0]*a[1];
-
-          test = n[0]*c[0]+n[1]*c[1]+n[2]*c[2];
-          if (test < 0.0) {
-            bound[iMarker][iElem_Surface]->Change_Orientation();
-            triangle_flip++;
-          }
+        /*--- The normal of the triangle formed by the line and domain
+         * point should point in the positive z direction. ---*/
+        constexpr int nDim = 2;
+        su2double a[nDim]={0.0}, b[nDim]={0.0};
+        GeometryToolbox::Distance(nDim, Coord_2, Coord_1, a);
+        GeometryToolbox::Distance(nDim, Coord_3, Coord_1, b);
+        bool test = a[0]*b[1]-a[1]*b[0] < 0.0;
```

Comment:
this looks like checkTria - consider moving that lambda to a small function too so you can reuse if possible

Reply:
Deleting 50% of duplicated code is not good enough?